crates/allocator/src/bump.rs

```diff
@@ -117,3 +170,33 @@ impl InnerAlloc {
         alloc_start as *mut u8
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn can_alloc_a_box() {
+        let _b = Box::new(1);
+    }
+
+    #[test]
+    fn can_alloc_a_vec() {
+        let mut v = Vec::new();
+        v.push(1)
+    }
+
+    #[test]
+    fn can_alloc_a_big_vec() {
+        let mut v = Vec::with_capacity(PAGE_SIZE);
+        v.push(true)
+    }
```

Comment:
I would write the tests in terms of unit tests, using the `BumpAllocator` directly instead of the `#[global_alloc]`.

Reply:
Then you can also specify specific testing APIs which you can call (if you think it is needed) to grasp more information about what happens behind the scene.

Reply:
This was a really helpful tip actually. After banging my head trying to hunt down segfaults today this allowed me to sidestep it all